### PR TITLE
Issue/3232 viewbinding product detail p1

### PR DESCRIPTION
**Pull request by nbradbury**
This is the first of several PRs converting product detail to `ViewBinding`. In this PR, I focused on files in the main `ui/products` folder.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductShippingClassFragment.kt [hunk 1]
```diff
@@ -39,25 +38,27 @@ class ProductShippingClassFragment : BaseFragment(), ShippingClassAdapterListene
 
     private var shippingClassAdapter: ProductShippingClassAdapter? = null
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        return inflater.inflate(R.layout.fragment_product_shipping_class_list, container, false)
-    }
+    private var _binding: FragmentProductShippingClassListBinding? = null
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductShippingClassFragment.kt:41]: Missing the logic to set `_binding = null` in `onDestroyView()` - was this intentional?

**Review by AmandaRiu — COMMENTED** [review]
@nbradbury Looks good! Left a couple minor code comments for your review.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductShippingClassFragment.kt (hunk 1 above)
Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductShippingClassFragment.kt:41]: @AmandaRiu Thanks for catching that - fixed in 52fe307, ready for another round.

**Review by AmandaRiu — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**